redis.go

```diff
@@ -113,11 +111,20 @@ func blocking(
 		c.WriteInline("QUEUED")
 		return
 	}
+
+	cleanup := make(chan struct{})
+	defer close(cleanup)
+	timedOut := false
 	if timeout != 0 {
-		dl = time.NewTimer(timeout)
-		defer dl.Stop()
-		dlc = dl.C
+		go setCondTimer(m.signal, &timedOut, timeout, cleanup)
```

Comment:
Why not pass `m.Ctx` instead of the `cleanup` channel? That channel could go away then ...

Reply:
Ah, no, sorry. I reckon that must be a different context. But still, I think a cancellable context is nicer than a channel, because the local context could be derived from the global one, so checking for only one of them is sufficient.